src/cpp/rtps/DataSharing/DataSharingListener.cpp

```diff
@@ -133,12 +133,11 @@ void DataSharingListener::process_new_data ()
 {
     EPROSIMA_LOG_INFO(RTPS_READER, "Received new data notification");
 
-    std::unique_lock<std::mutex> lock(mutex_);
-
-    // It is safe to 'forget' any change now
-    notification_->notification_->new_data.store(false);
-    // All places where this is set to true is locked by the same mutex, memory_order_relaxed is enough
-    writer_pools_changed_.store(false, std::memory_order_relaxed);
+    notification_->notification_->new_data.store(false,std::memory_order_release);
+    // Using Acquire-Release semantics can avoid some blocking problems in traditional locking, such as deadlock and priority inversion. 
+    // This usually means higher concurrency performance because threads don't have to wait for locks to be released.
+    // Through Acquire-Release, you can achieve finer-grained control over specific variables or data structures without having to lock the entire resource or object, which can reduce contention and improve concurrency.
+    writer_pools_changed_.store(false,std::memory_order_release);
 
     // Loop on the writers looking for data not read yet
     for (auto it = writer_pools_.begin(); it != writer_pools_.end(); ++it)
```

Comment:
If there was a problem with effectiveness here, it would have been exposed before. I'll keep an eye on this later, and if there's a problem I'll fix it

Reply:
The places where `writer_pools_` is modified are protected by taking `mutex_`.
Your PR is removing the guard on the `mutex_` during the traversal of `writer_pools_` here.

Reply:
> The places where `writer_pools_` is modified are protected by taking `mutex_`. Your PR is removing the guard on the `mutex_` during the traversal of `writer_pools_` here.

Don't worry about this, I use the lock_free mechanism here. It is thread safe.

Reply:
`writer_pools_` is a `std::vector<WriterInfo>` which is not thread-safe. If `DataSharingListener::remove_datasharing_writer()` is called on one thread while another thread is inside `DataSharingListener::process_new_data()`, an invalidated iterator on the vector could be dereferenced

Reply:
you are right. But I tested it many times under multi-thread conditions and there was no problem.

Reply:
I re-locked